### PR TITLE
Add a function to get config data

### LINKED ISSUE #3: Add a Jupyter Config loader
Add a function that retrieves a config value from the HTML page via a handlebar template in a JSON script tag.

```
<script id="jupyter-config-data" type="application/json">
{org: 10, items:["one","two"]}
</script>
```

`element = document.getElementById('jupyter-config-data);`

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #3.

### CODE REVIEW

**Comment by blink1073** [comment]
cc @minrk

File: src/index.ts [hunk 1]
```diff
@@ -233,3 +233,30 @@ class PromiseDelegate<T> {
...
+ * @returns The config value or `undefined` if not found.
+ *
+ * #### Notes
+ * For browser based applications, it is assumed that the page HTML
+ * includes a script tag with the id `jupyterConfigData` containing the
+ * configuration as valid JSON.
+ */
+export
+function getConfigOption(name: string): any {
+  if (typeof window === 'undefined') {
```
Comment (sccolbert) [inline src/index.ts]: why would this run in node?

File: src/index.ts [hunk 2]
```diff
@@ -233,3 +233,30 @@ class PromiseDelegate<T> {
...
+    // TODO: support node here.
+    return void 0;
+  }
+  let el = document.getElementById('jupyterConfigData');
```
Comment (sccolbert) [inline src/index.ts]: what the style guidelines for ids? I prefer hyphenated names to distinguish them from variables. But, just a preference.

File: src/index.ts (hunk 1 above)
Comment (blink1073) [inline src/index.ts]: jupyter-js-services supports node, it was a requested feature.

File: src/index.ts (hunk 2 above)
Comment (blink1073) [inline src/index.ts]: Google style guide says hyphenated: https://google.github.io/styleguide/htmlcssguide.xml#ID_and_Class_Name_Delimiters

File: src/index.ts [hunk 3]
```diff
@@ -233,3 +233,30 @@ class PromiseDelegate<T> {
...
+  if (!el) {
+    return void 0;
+  }
+  let config = JSON.parse(el.textContent);
+  return (config as any)[name];
```
Comment (sccolbert) [inline src/index.ts]: `JSON.parse` returns `any`. The cast isn't needed.

We can assume the config data in the node to be static, so we only need to parse once and store the parsed result in an internal module variable.

We should also define some type-safe overloads of `getConfigOption` with the string literals for the known config options. i.e.

``` typescript
function getConfigOption(name: 'baseUrl'): string;
function getConfigOption(name: 'port'): number;
function getConfigOption(name: 'whatever'): boolean;
function getConfigOption(name: string): any {
  // implementation
}
```

Comment (blink1073) [inline src/index.ts]: If the config is nested, we'd have to make a stringify/parse before giving it away, but I suppose that is cheaper than parsing the whole thing every time.

**Comment by sccolbert** [comment]
Is this the right repo for this? It seems a bit out of place wrt to the others.

File: src/index.ts (hunk 3 above)
Comment (sccolbert) [inline src/index.ts]: are you worried about mutations? Just recursive freeze the whole object have parsing it the first time.

**Comment by sccolbert** [comment]
It seems like this should live as part of the core browser libs.

**Comment by blink1073** [comment]
This is a copy of `jupyter-js-services/src/utils.ts`.  The intent was to be a `domutils` like functionality to be used across the Jupyter-JS libs.

**Comment by sccolbert** [comment]
But you are already running into competing requirements. This function relies on a browser runtime, but some people want to use the other things in it from within node (aside: why?). 

Jupyter-js-services itself doesn't need this. The code _calling_ jupyter-js-service does, because they have to provide the base url. To me, this function would make more sense there.

**Comment by blink1073** [comment]
The node requirement was for the kernel-proxy service. 

I see this as a shared lib to be used by multiple other libraries.  It could be that some of the functions are only needed by jupyter-js-services, and should be removed here.

**Comment by sccolbert** [comment]
Yeah. I get the point about utils. But we probably want to make the distinction between vanilla JS utilities, and utilites which make assumptions about the runtime and global state (like this one).

**Comment by blink1073** [comment]
Something like `jupyter-js-browser-utils`, `jupyter-js-utils`, and `jupyter-js-node-utils`?

**Comment by sccolbert** [comment]
Let's see what @minrk thinks about it

**Comment by minrk** [comment]
Unless there's an installation/dependency conflict, I think it's fine for this to be a single repo. If there is or there's another reason why having them together causes an increase in the complexity of this package and/or those that use it, that's fine too. But if it's just because this function would only be used in the browser, I don't think that is enough to need another repo.

**Pull request merged**